### PR TITLE
Remove duplicate lines in the generated code

### PR DESCRIPTION
**Pull request by kunichik**

### CODE REVIEW

File: codegen/facelift/templates/QMLAdapter.template.h [hunk 1]
```diff
@@ -48,7 +48,9 @@
 #include "{{interfaceName}}.h"
 
 {% for type in interface.referencedTypes %}
-{{type.requiredInclude}}
+    {% if type.requiredInclude != "" %}
+        {{type.requiredInclude}}
```
Comment (jacky309) [inline codegen/facelift/templates/QMLAdapter.template.h:52]: this line does not need any indentation

File: codegen/facelift/templates/Service.template.h [hunk 2]
```diff
@@ -43,7 +43,9 @@
 
 // Dependencies
 {% for type in interface.referencedTypes %}
-{{type.requiredInclude}}
+    {% if type.requiredInclude != "" %}
+        {{type.requiredInclude}}
```
Comment (jacky309) [inline codegen/facelift/templates/Service.template.h:47]: this line does not need any indentation

File: codegen/facelift/templates/Struct.template.h [hunk 3]
```diff
@@ -44,7 +44,9 @@
 
 // Dependencies
 {% for field in struct.fields %}
-{{field.type.requiredInclude}}
+    {% if field.type.requiredInclude != "" %}
+        {{field.type.requiredInclude}}
```
Comment (jacky309) [inline codegen/facelift/templates/Struct.template.h:48]: this line does not need any indentation

File: codegen/facelift/templates/StructQObjectWrapper.template.h [hunk 4]
```diff
@@ -44,7 +44,9 @@
 
 // Dependencies
 {% for field in struct.fields %}
-{{field.type.requiredInclude}}
+    {% if field.type.requiredInclude != "" %}
+        {{field.type.requiredInclude}}
```
Comment (jacky309) [inline codegen/facelift/templates/StructQObjectWrapper.template.h:48]: this line does not need any indentation

File: codegen/facelift/templates/Struct.template.h [hunk 5]
```diff
@@ -44,7 +44,9 @@
 
 // Dependencies
 {% for field in struct.fields %}
```
Comment (jacky309) [inline codegen/facelift/templates/Struct.template.h:46]: Did you try the following ?
```suggestion
{% for field in struct.fields -%}
{{field.type.requiredInclude}}
{-% endfor %}
```

If that works, I would prefer that approach.

Comment (kunichik) [reply codegen/facelift/templates/Struct.template.h:46]: Good idea, but unfortunately it doesn't works, there is such code generated:

 #include "facelift/example/addressbook/SubInterface.h"#include "facelift/example/addressbook/Contact.h"#include "facelift/example/addressbook/FailureReason.h"

**Review by bitmouse — APPROVED** [review]

**Pull request merged**